### PR TITLE
[5.1] Convert mod_banners to service provider

### PR DESCRIPTION
**Pull request by heelc29**
### Summary of Changes
 
Converts the banners site module to service provider

### Testing Instructions

- create site module `Banners`
- check site output

Note: system test exists
https://github.com/joomla/joomla-cms/blob/5.1-dev/tests/System/integration/site/modules/mod_banners/Default.cy.js

### Actual result BEFORE applying this Pull Request

module works
image

### Expected result AFTER applying this Pull Request

module works
image

### Link to documentations
Please select:
- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: modules/mod_banners/src/Helper/BannersHelper.php [hunk 1]
```diff
@@ -31,13 +31,17 @@ class BannersHelper
      * Retrieve list of banners
      *
      * @param   Registry        $params  The module parameters
-     * @param   BannersModel    $model   The model
      * @param   CMSApplication  $app     The application
      *
      * @return  mixed
```
Comment (sandewt) [inline modules/mod_banners/src/Helper/BannersHelper.php:36]:
```suggestion
     * @return  array
```

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 57e05572ef713f4f10e37640b8081f375e498f6f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42214">issues.joomla.org/tracker/joomla-cms/42214</a>.</sub>

File: modules/mod_banners/src/Helper/BannersHelper.php (hunk 1 above)
Comment (heelc29) [reply modules/mod_banners/src/Helper/BannersHelper.php:36]: Return type of `getItems()` is mixed (specific `array|bool`)
https://github.com/joomla/joomla-cms/blob/c9b318106e9b34b3d736f3bd2c8c04968701bcfa/libraries/src/MVC/Model/ListModel.php#L252-L259

Comment (sandewt) [reply modules/mod_banners/src/Helper/BannersHelper.php:36]: The relevant file that is loaded is the following: 
Return type of `getItems()` is an array (specific `array` and not a bool). Correct?

https://github.com/joomla/joomla-cms/blob/bb8e5278edf0244ccfc739c12b1425a855e5b2e1/components/com_banners/src/Model/BannersModel.php#L248-L256

Also see?
`use Joomla\Component\Banners\Site\Model\BannersModel;`

Comment (sandewt) [reply modules/mod_banners/src/Helper/BannersHelper.php:36]: It overrides `ListModel::getItems`. Correct?

**Comment by LadySolveig** [comment]
Thank you @heelc29 and for testing and review @ceford @viocassel @sandewt

**Pull request merged**